test/integration/test_notebooks_in_dss_docker_image.py

```diff
@@ -0,0 +1,73 @@
+import io
+import os
+from inspect import cleandoc
+from pathlib import Path
+
+import pytest
+
+from test.integration.docker.exec_run import exec_command
+from test.integration.docker.image import image
+from test.integration.docker.in_memory_build_context import InMemoryBuildContext
+from test.integration.docker.container import container
+
+TEST_RESOURCE_PATH = Path(__file__).parent.parent / "notebooks"
+
+
+@pytest.fixture(scope="session")
+def notebook_test_dockerfile_content(dss_docker_image) -> str:
+    yield cleandoc(
+        f"""
+        FROM {dss_docker_image.image_name} 
+        COPY notebooks/* /tmp/notebooks/
+        RUN cp /tmp/notebooks/* "$NOTEBOOK_FOLDER_INITIAL"
+        WORKDIR $NOTEBOOK_FOLDER_INITIAL
+        RUN "$VIRTUAL_ENV/bin/python3" -m pip install nbclient nbformat pytest
```

Comment:
Will do, and I probably extract the dependencies into a file which gets copied into the image, this should make updating easier

Reply:
Should we have a function similar `test_notebook` that runs the pytest-check-links for all the notebooks? Or  for all directories with notebooks, since we can specify the source as directory/*.ipynb. But I don't think it can do a deep search.

Reply:
done